File: InvoiceOCRApp/DTOs/InvoiceOcrResultDto.cs
using System.Collections.Generic;

namespace InvoiceOCRApp.DTO;

public class InvoiceOcrResultDto
{
    public int? Id { get; set; }
    public string InvoiceNumber { get; set; }
    public DateOnly? InvoiceDate { get; set; }
    public string CustomerName { get; set; }
    public decimal? TotalAmount { get; set; }
    public decimal? VAT { get; set; }
    public decimal? Subtotal { get; set; }
    public List<InvoiceOcrItemDto> Items { get; set; } = new();
    public string? RawText { get; set; }
    public double? OcrConfidence { get; set; }
}

File: invoice-ocr-app/src/app/models/invoice-ocr-result.dto.ts
export interface InvoiceItemDto {
  description: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export interface InvoiceOcrResultDto {
  id : number | null;
  invoiceNumber: string;
  invoiceDate: string | null;
  customerName: string;
  totalAmount: number | null;
  vat: number | null;
  rawText: string | null;
  items: InvoiceItemDto[];
}
